### PR TITLE
Fixs for klics

### PR DESCRIPTION
**Pull request by clienthax**
Base commit for Jarves to add SCE header support to the decrypter

### CODE REVIEW

**Comment by clienthax** [comment]
Tested with: Castlevania: Harmony of Despair
Old code would fail out as soon as the key was wrong / as soon as verify failed
Game seems to try multiple keys, with one of them being correct.

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 1]
```diff
@@ -65,15 +65,21 @@ s32 npDrmIsAvailable(vm::cptr<u8> k_licensee_addr, vm::cptr<char> drm_path)
 	u32 magic;
 	enc_file.read<u32>(magic);
 	enc_file.seek(0);
-	if (magic != "NPD\0"_u32)
+	if (magic == "SCE\0"_u32) {
```
Comment (hcorion) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp]: Curly bracket on next line.

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 2]
```diff
@@ -65,15 +65,21 @@ s32 npDrmIsAvailable(vm::cptr<u8> k_licensee_addr, vm::cptr<char> drm_path)
...
+	else if (magic != "NPD\0"_u32)
 	{
 		// for now assume its just unencrypted
 		sceNp.notice("npDrmIsAvailable(): Assuming edat file is unencrypted at %s", enc_drm_path);
 		return CELL_OK;
 	}
 
 	sceNp.warning("npDrmIsAvailable(): Found DRM license file at %s", enc_drm_path);
-	sceNp.warning("npDrmIsAvailable(): Using k_licensee 0x%s", k_licensee_str);
+	if (k_licensee_addr) {
```
Comment (hcorion) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp]: ...and here...

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 3]
```diff
@@ -84,22 +90,24 @@ s32 npDrmIsAvailable(vm::cptr<u8> k_licensee_addr, vm::cptr<char> drm_path)
 	auto edatkeys = fxm::get_always<EdatKeys_t>();
 
 	// Search for a compatible RAP file. 
-	for (const auto& entry : fs::dir(rap_lpath))
-	{
-		if (entry.name.find(title_id) != -1)
+	if (!k_licensee_addr) {
```
Comment (hcorion) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp]: ...and here...

Comment (clienthax) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp]: Honestly im not sure why you are flagging these all up when its a base commit for another pr..

**Pull request closed without merging**